python/hsfs/engine/hive.py

```diff
@@ -249,45 +253,19 @@ def save_dataframe(
 
         # Launch job
         print("Launching ingestion job...")
-        self._job_api.launch(ingestion_job.job.name)
+        fg_job = self._job_api.launch(ingestion_job.job.name)
         print(
             "Ingestion Job started successfully, you can follow the progress at {}".format(
                 self._get_job_url(ingestion_job.job.href)
             )
         )
 
-    def _get_job_url(self, href: str):
-        """Use the endpoint returned by the API to construct the UI url for jobs
+        # If the user passed the wait_for_job option consider it,
+        # otherwise use the default True
+        if offline_write_options.pop("wait_for_job", True):
```

Comment:
you could move the `if` statement into the function and just use the boolean as condition in the while loop, so it's not repeated for training datasets and featuregroups

Reply:
done.